sql/sql_repl.cc

```diff
@@ -4052,7 +4052,10 @@ bool change_master(THD* thd, Master_info* mi, bool *master_info_added)
   if (lex_mi->connect_retry)
     mi->connect_retry = lex_mi->connect_retry;
   if (lex_mi->retry_count)
+  {
     mi->retry_count= lex_mi->retry_count;
+    mi->connects_tried= 0;
```

Comment:
Would it make sense to reset this for all `CHANGE MASTER` executions, rather than just when 
 MASTER_RETRY_COUNT` is provided? That is, if we are changing the connection parameters (e.g. `MASTER_HOST`), it would make sense to me that we shouldn't report the old `connects_tried` for a new connection.

Reply:
I say naw, because other parameters, including `MASTER_HOST` and co., don’t relate to `Connects_Tried`, and so have no reason to clear it when it’s intended to remain past STOP REPLICA.